github-api.php

```diff
@@ -3201,71 +3201,388 @@ function vipgoci_github_prs_commits_list(
+			}
+
+			/*
+			 * Remove possible git stuff in paths
+			 */
+			if ( $cur_file_paths_cleaned === false ) {
+				if ( 
+					( 0 === strpos( $cur_file, 'a/' ) ) ||
+					( 0 === strpos( $cur_file, 'b/' ) )
+				) {
+					$cur_file = substr(
+						$cur_file,
+						2
+					);
+				}
+
+				if ( 0 === strpos( $cur_file_minus, 'a/' ) ) {
+					$cur_file_minus = substr(
+						$cur_file_minus,
+						2
+					);
+				}
+
+				if ( 0 === strpos( $cur_file_plus, 'b/' ) ) {
+					$cur_file_plus = substr(
+						$cur_file_plus,
+						2
+					);
+				}
+
+				$cur_file_paths_cleaned = true;
+			}
+
+
+			/*
+			 * Logic to handle file names.
+			 */
+			if (
+				( ! empty( $cur_file_minus ) ) ||
+				( ! empty( $cur_file_plus ) )
+			) {
+				if ( $cur_file_minus === $cur_file_plus ) {
+					$cur_file = $cur_file_plus;
+					$cur_file_status = 'modified';
+				}
+
+				else {
+					/*
+					 * If the file names do not match,
+					 * and either is /dev/null, then
+					 * it is a new file or a removed file.
+					 */
+					if ( $cur_file_minus === '/dev/null' ) {
+						$cur_file = $cur_file_plus;
+						$cur_file_status = 'added';
+					}
+
+					else if ( $cur_file_plus === '/dev/null' ) {
+						$cur_file = $cur_file_minus;
+						$cur_file_status = 'removed';
+					}
+
+					/*
+					 * No match and no mention of /dev/null,
+					 * so the file must have been renamed.
+					 */
+					else {
+						$cur_file = $cur_file_plus;
+						$cur_file_status = 'renamed';
+					}
+				}
+			}
+
+			/*
+			 * If the file is not yet registered in the
+			 * 'diff' array, do so now. This will include
+			 * files that only have permission changes.
+			 */
+			if (
+				( null !== $cur_file ) &&
+				( null !== $cur_file_status ) &&
+				( ! isset( $diff_results['files'][ $cur_file ] ) )
+			) {
+				$diff_results['files'][ $cur_file ] = array(
+					'filename'	=> $cur_file,
+					'patch'		=> '',
+					'status'	=> $cur_file_status,
+					'additions'	=> 0,
+					'deletions'	=> 0,
+					'changes'	=> 0,
+				);
+			}
+
+			if ( null !== $cur_file_status ) {
+				/*
+				 * Update file-status each time we loop
+				 * as the calculated status might change.
+				 */
+				$diff_results['files'][ $cur_file ][ 'status' ] =
+					$cur_file_status;
+			}
+
+			if ( 'patch' !== $cur_mode ) {
+				/* Not in patch mode, so do not collect patch */
+				continue;
+			}
+
+			/*
+			 * Sanity check.
+			 */
+			if ( empty( $cur_file ) ) {
+				vipgoci_log(
+					'Problem when getting git diff, no file name found in patch',
+					array(
+						'local_git_repo'	=> $local_git_repo,
+						'commit_id_a'		=> $commit_id_a,
+						'commit_id_b'		=> $commit_id_b,
+						'cur_file'		=> $cur_file,
+						'cur_file_minus'	=> $cur_file_minus,
+						'cur_file_plus'		=> $cur_file_plus,
+
+					)
+				);
+
+				continue;
+			}
+
+			/*
+			 * Keep statistics on lines changed (if any).
+			 */
+			if ( strlen( $git_result_item ) > 0 ) {
+				if ( '+' === $git_result_item[0] ) {
```

Comment:
Suggestion. 
The + => additions could be a constant defined previously, and based on that we could refactor remove the [if/else condition](https://github.com/Automattic/vip-go-ci/pull/153/files#diff-4a0fcb15c05fb7f32eb99b19cb448971027c14ca8f088625a92a9e537795c0ffR3530-R3538):

```
const GIT_DIFF_OPTIONS = array ('+' => 'additions', '-' => 'deletions');

$diff_results['files'][ $cur_file ][GIT_DIFF_OPTIONS[$git_result_item[0]]]++;
$diff_results['files'][ $cur_file ]['changes']++;

```

Reply:
I agree, I think the constant is a good idea.

Reply:
Fixed in b25958aebec7da25049c5ff9471cebfe65b51ade